ast/src/parsed/display.rs

```diff
@@ -101,6 +55,38 @@ impl<T: Display> Display for ASMStatement<T> {
                 }
             ),
             ASMStatement::Label(_, name) => write!(f, "{name}::"),
+            ASMStatement::Degree(_, degree) => write!(f, "degree {};", degree),
+            ASMStatement::RegisterDeclaration(_, name, flag) => write!(
+                f,
+                "reg {}{};",
+                name,
+                flag.as_ref()
+                    .map(|flag| format!("[{flag}]"))
+                    .unwrap_or_default()
+            ),
+            ASMStatement::InstructionDeclaration(_, name, params, body) => {
+                write!(
+                    f,
+                    "instr {}{} {{{}}}",
+                    name,
+                    params,
+                    body.iter()
+                        .map(|e| format!("{e}"))
+                        .collect::<Vec<_>>()
+                        .join(" ")
+                )
+            }
+            ASMStatement::InlinePil(_, statements) => {
+                write!(
+                    f,
+                    "pil{{\n{}\n}}",
+                    statements
+                        .iter()
+                        .map(|s| format!("{}", s))
+                        .collect::<Vec<_>>()
+                        .join("\n")
+                )
+            }
```

Comment:
Why were these moved around? Looks like almost the same content?

Reply:
good call will fix

Reply:
fine as is, just wanted to know if there was more to it

Reply:
There isnt, I changed it back the diff should be smaller